src/core/alt-table.rkt

```diff
@@ -174,33 +158,32 @@
     (if (*pareto-mode*)
         (hash-ref (alt-table-alt->cost atab) a)
         (backup-alt-cost a)))
-
-  (argmax alt-cost (argmins alt-num-points (argmins alt-done? altns))))
+  ;; Rank by multiple metrics
+  (define not-done (argmins alt-done? (set->list removable)))
+  (define least-best-points (argmins alt-num-points not-done))
+  (define worst-cost (argmaxs alt-cost least-best-points))
+  ;; The set may have non-deterministic behavior,
+  ;; so we can only rely on some total order
+  (first (order-altns worst-cost)))
```

Comment:
Interested to know if this slows it down. I believe I tried it once or twice and it seemed slower. Still probably worth it but let's check.

Reply:
Probably I am over-reacting because I spent like three days trying to optimize this loop into the ground. IIRC minimizing the alt-table isn't a significant use of time, so I'm not actually to worried about this.